code/game/machinery/iv_drip.dm

```diff
@@ -166,23 +175,46 @@
 		return
 	if(attached)
 		visible_message("<span class='notice'>[attached] is detached from [src].</span>")
-		attached = null
-		update_icon()
+		detach_IV()
 		return
 	else if(beaker)
 		eject_beaker(user)
 	else
 		toggle_mode()
 
-/obj/machinery/iv_drip/AltClick(mob/living/user)
-	if(!user.canUseTopic(src, be_close=TRUE))
-		return
+/obj/machinery/iv_drip/attack_hand_secondary(mob/user, modifiers)
 	if(dripfeed)
 		dripfeed = FALSE
 		to_chat(usr, "<span class='notice'>You loosen the valve to speed up the [src].</span>")
 	else
 		dripfeed = TRUE
 		to_chat(usr, "<span class='notice'>You tighten the valve to slowly drip-feed the contents of [src].</span>")
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+///called when an IV is attached
+/obj/machinery/iv_drip/proc/attach_IV(mob/living/target, mob/usr)
```

Comment:
```suggestion
/obj/machinery/iv_drip/proc/attach_iv(mob/living/target, mob/user)
```

Do not shorten variable names: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names